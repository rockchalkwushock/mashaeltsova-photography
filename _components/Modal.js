import Photo from './Photo'

const Modal = ({ id, onDismiss }) => (
  <div onClick={onDismiss}>
    <Photo id={id} />
  </div>
)

export default Modal
