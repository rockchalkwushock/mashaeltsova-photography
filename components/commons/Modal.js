import FontAwesome from 'react-fontawesome'
import PropTypes from 'prop-types'

import { styles } from '../../lib'

const Modal = ({ msg, status }) => {
  let icon
  if (status === 'error') {
    icon = <FontAwesome name="times-circle" size="3x" />
  } else if (status === 'loading') {
    icon = <FontAwesome name="spinner" pulse size="3x" />
  } else if (status === 'success') {
    icon = <FontAwesome name="check-circle" size="3x" />
  }
  return (
    <div className={status}>
      {icon}
      <span>{msg}</span>
      <style jsx>{`
        div {
          background-color: ${styles.colors.white};
          display: flex;
          flex-direction: column;
          padding: 0.5em;
        }
        div > * {
          padding: 0.25em;
          flex: 1 1 auto;
        }
        .error {
          color: red;
        }
        .loading {
          color: ${styles.colors.warm};
        }
        .success {
          color: green;
        }
        span {
          color: ${styles.colors.charcoal};
        }
      `}</style>
    </div>
  )
}

Modal.propTypes = {
  msg: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
}

export default Modal
