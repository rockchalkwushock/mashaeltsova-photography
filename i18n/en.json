{
  "buttonBooking": "Book Now!",
  "buttonGallery": "View my work here!",
  "formEmail": "Email",
  "formFirstName": "First Name",
  "formLastName": "Last Name",
  "formMessage": "Short Message",
  "formPhone": "Phone",
  "formErrorEmailPattern": "Not a valid email.",
  "formErrorNamePattern":
    "Not a valid pattern. May only contain upper and lowercase letters.",
  "formErrorMessagePattern":
    "May only contain upper & lower case letters and numbers.",
  "formErrorPhonePattern":
    "Should be match one of the following: +11231231234, 11231231234, 1231231234,",
  "formErrorRequired": "Please fill this in.",
  "formErrorTooLong": "Max length 140 characters.",
  "formErrorTooShort": "Minimum length 2 characters.",
  "formRequiredMessage": "All fields required.",
  "galleryNav": "Family,Portrait,Travel,Wedding",
  "modalError": "I'm sorry there seems to be a problem, please try back later.",
  "modalLoading": "Your request is being sent now.",
  "modalSuccess": "Your booking inquiry has been sent!",
  "sectionAbout":
    "For many years now I have had a great passion for capturing a moment in time and sharing the beauty of that moment with the world. I enjoy taking pictures for many varying events. When I am not doing photo shoots you will find me preforming English tutoring, or practicing my other passion, surfing.",
  "sectionPlug":
    "Contact me if you want to relive those precious moments for years to come!",
  "sectionServices":
    "I am willing to negotiate other requested services, however the below are what I primarily offer: Baby Portraits,Family Portraits,Self-Portraits,Engagements,Weddings",
  "sideNav": "Home,Gallery"
}
