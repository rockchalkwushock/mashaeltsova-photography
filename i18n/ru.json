{
  "buttonBooking": "Забронировать сейчас!",
  "buttonGallery": "Мои работы можно посмотреть тут!",
  "formEmail": "Электронный адрес",
  "formFirstName": "Имя",
  "formLastName": "Фамилия",
  "formMessage": "Сообщение",
  "formPhone": "Телефон",
  "formErrorEmailPattern": "FIXME",
  "formErrorNamePattern": "FIXME",
  "formErrorMessagePattern":
    "Может содержать заглавные и строчные буквы, цифры, и .!?,;'",
  "formErrorPhonePattern":
    "Должно соответствовать одному из следующих: +1-234-567-8901, +61-234-567-89-01 +46-234 5678901",
  "formErrorRequired": "Пожалуйста, заполните это поле.",
  "formErrorTooLong": "Максимальная длина 140 знаков.",
  "formErrorTooShort": "минимальная длина 2 знака.",
  "formRequiredMessage": "Необходимо заполнить все поля.",
  "galleryNav": "Семья,Портрет,Путешествия,Свадьбы",
  "modalError": "Произошла ошибка, попробуйте еще раз.",
  "modalLoading": "Ваш запрос отправляется.",
  "modalSuccess": "Ваш запрос на фотосессию отправлен!",
  "sectionAbout":
    "Уже много лет я занимаюсь фотографией и мне нравится моя работа. Мне нравится общаться и фотографировать людей. Если я не фотографирую то, путешествую или учу английский язык, или просто наслаждаюсь жизнью.",
  "sectionPlug": "Связитесь со мной , если хотите обсудить съемку!",
  "sectionServices":
    "Открыта к обсуждению съемок, специализируюсь на следующих съемках: Детская и семейная съемка,Портреты,Помолвки,Свадьбы",
  "sideNav": "FIXME"
}
