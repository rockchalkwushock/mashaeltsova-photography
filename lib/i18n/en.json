{
  "description": "Photography Business Website for Masha Eltsova Photography",
  "greeting": "Hello World!",
  "title": "Masha Eltsova Photography"
}