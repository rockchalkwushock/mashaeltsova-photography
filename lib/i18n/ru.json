{
  "book": "Забронируйте сейчас!",
  "description": "Сайт фотографии для Маши Ельцовой",
  "email": "Эл. адрес",
  "emailInvalid": "Неверный адрес электронной почты.",
  "emailRequired": "Требуется электронная почта.",
  "errorResponse":
    "У вас возникла проблема с вашим запросом, пожалуйста, попробуйте позже.",
  "firstName": "Имя",
  "firstNameRequired": "Имя требуется.",
  "firstNameInvalid": "Неверное имя. Только альфа-символы.",
  "firstNameLength": "Имя должно быть не менее 2 символов.",
  "greeting": "Привет мир!",
  "lastName": "Фамилия",
  "lastNameInvalid": "Недопустимое имя. Только альфа-символы.",
  "lastNameLength": "Фамилия должна быть не менее 2 символов.",
  "lastNameRequired": "Фамилия обязательна.",
  "message": "Каковы ваши потребности в бронировании?",
  "phone": "Номер телефона",
  "phoneInvalid":
    "Неправильный номер телефона. Принимает: +1-234-567-8901, +61-234-567-89-01.",
  "phoneRequired": "Требуется номер телефона.",
  "successResponse": "Ваш запрос на бронирование был отправлен!",
  "title": "Маша Ельцова Фотография",
  "messageRequired": "Сообщение требуется.",
  "messageInvalid": "FIXME",
  "messageMaxLength": "Сообщение должно быть не более 140 символов.",
  "messageMinLength": "Сообщение должно содержать не менее 10 символов.",
  "rcws": "Разработка RCWS"
}
