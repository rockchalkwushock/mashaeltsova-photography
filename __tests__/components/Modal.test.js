import React from 'react'
import { render } from 'enzyme'

import Modal from '../../components/commons/form/FormModal'
import en from '../../i18n/en.json'
import ru from '../../i18n/ru.json'

describe('Component: Modal', () => {
  describe('Error State', () => {
    test('Renders without exploding in English', () => {
      const tree = render(<Modal msg={en.modalError} status="error" />)
      expect(tree).toMatchSnapshot()
    })
    test('Renders without exploding in Russian', () => {
      const tree = render(<Modal msg={ru.modalError} status="error" />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Loading State', () => {
    test('Renders without exploding in English', () => {
      const tree = render(<Modal msg={en.modalLoading} status="loading" />)
      expect(tree).toMatchSnapshot()
    })
    test('Renders without exploding in Russian', () => {
      const tree = render(<Modal msg={ru.modalLoading} status="loading" />)
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Success State', () => {
    test('Renders without exploding in English', () => {
      const tree = render(<Modal msg={en.modalSuccess} status="success" />)
      expect(tree).toMatchSnapshot()
    })
    test('Renders without exploding in Russian', () => {
      const tree = render(<Modal msg={ru.modalSuccess} status="success" />)
      expect(tree).toMatchSnapshot()
    })
  })
})
