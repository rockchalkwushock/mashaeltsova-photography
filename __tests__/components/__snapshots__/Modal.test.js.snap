// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Modal Error State Renders without exploding in English 1`] = `
<div
  class="jsx-2863241787 error"
>
  <span
    aria-hidden="true"
    class="fa fa-times-circle fa-3x"
  />
  <span
    class="jsx-2863241787"
  >
    I'm sorry there seems to be a problem, please try back later.
  </span>
</div>
`;

exports[`Component: Modal Error State Renders without exploding in Russian 1`] = `
<div
  class="jsx-2863241787 error"
>
  <span
    aria-hidden="true"
    class="fa fa-times-circle fa-3x"
  />
  <span
    class="jsx-2863241787"
  >
    Произошла ошибка, попробуйте еще раз.
  </span>
</div>
`;

exports[`Component: Modal Loading State Renders without exploding in English 1`] = `
<div
  class="jsx-2863241787 loading"
>
  <span
    aria-hidden="true"
    class="fa fa-spinner fa-3x fa-pulse"
  />
  <span
    class="jsx-2863241787"
  >
    Your request is being sent now.
  </span>
</div>
`;

exports[`Component: Modal Loading State Renders without exploding in Russian 1`] = `
<div
  class="jsx-2863241787 loading"
>
  <span
    aria-hidden="true"
    class="fa fa-spinner fa-3x fa-pulse"
  />
  <span
    class="jsx-2863241787"
  >
    Ваш запрос отправляется.
  </span>
</div>
`;

exports[`Component: Modal Success State Renders without exploding in English 1`] = `
<div
  class="jsx-2863241787 success"
>
  <span
    aria-hidden="true"
    class="fa fa-check-circle fa-3x"
  />
  <span
    class="jsx-2863241787"
  >
    Your booking inquiry has been sent!
  </span>
</div>
`;

exports[`Component: Modal Success State Renders without exploding in Russian 1`] = `
<div
  class="jsx-2863241787 success"
>
  <span
    aria-hidden="true"
    class="fa fa-check-circle fa-3x"
  />
  <span
    class="jsx-2863241787"
  >
    Ваш запрос на фотосессию отправлен!
  </span>
</div>
`;
